src/Workspaces/Core/Portable/CodeActions/CodeActionPriority.cs

```diff
@@ -2,25 +2,29 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Microsoft.CodeAnalysis.Diagnostics;
+
 namespace Microsoft.CodeAnalysis.CodeActions
 {
+#pragma warning disable CA1200 // Avoid using cref tags with a prefix
+    /// <summary>
+    /// Internal priority used to bluntly place items in a light bulb in strict orderings.  Priorities take
+    /// the highest precedence when ordering items so that we can ensure very important items get top prominance,
+    /// and low priority items do not.
+    /// </summary>
+    /// <remarks>
+    /// If <see cref="CodeActionPriority.High"/> is used, the feature that specifies that value should 
+    /// implement and return true for <see cref="IBuiltInAnalyzer.IsHighPriority"/>,
+    /// <see cref="T:Microsoft.CodeAnalysis.CodeFixes.CodeFixProvider.IsHighPriority"/> and
+    /// <see cref="T:Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringProvider.IsHighPriority"/>. This
+    /// will ensure that the analysis engine runs the providers that will produce those actions first,
+    /// thus allowing those actions to be computed and displayed prior to running all other providers.
+    /// </remarks>
     internal enum CodeActionPriority
     {
-        //
-        // Summary:
-        //     Lowest priority suggestion.
         Lowest = 0,
-        //
-        // Summary:
-        //     Low priority suggestion.
         Low = 1,
-        //
-        // Summary:
-        //     Medium priority suggestion.
         Medium = 2,
-        //
-        // Summary:
-        //     High priority suggestion.
         High = 3
```

Comment:
Should we have a check that asserts that if we end up with a high-priority action that the provider was marked as high priority? I worry we might do the former but not the latter should we ever add a new one and end up with a flaky experience.

(It's also possible you did add this and I missed it...)

Reply:
Seems reasonable!